src/s3.rs

```diff
@@ -12425,6 +12492,17 @@ impl<P, D> S3Client<P, D> where P: ProvideAwsCredentials, D: DispatchSignedReque
             None => host,
         }
     }
+
+    fn is_dns_compatible(&self, bucket: &BucketName) -> bool {
+        self.is_valid_subdomain(&bucket) && !bucket.contains(".")
+    }
+
+    fn is_valid_subdomain(&self, bucket: &BucketName) -> bool {
+        let subdomain_re: Regex = Regex::new(r"^[a-z0-9][a-z0-9.-]+[a-z0-9]$").unwrap();
```

Comment:
Fixed in  https://github.com/rusoto/rusoto/pull/344/commits/5dc9732dceec4abda20ecf9fb041ab3a6a1b410e
I also need to add lazy-static module to Cargo.toml